src/Correlate.DependencyInjection/IHttpClientBuilderExtensions.cs

```diff
@@ -35,6 +35,14 @@ public static IHttpClientBuilder CorrelateRequests(this IHttpClientBuilder build
             throw new ArgumentNullException(nameof(builder));
         }
 
+        builder.Services.AddOptions<CorrelationManagerOptions>()
+            .Configure((CorrelationManagerOptions cmo) =>
+            {
+                var options = new CorrelateClientOptions();
+                configureOptions(options);
+                cmo.LoggingScopeKey = options.LoggingScopeKey;
+            });
+
```

Comment:
There are some issues with this setup. In case someone registers two different HttpClients but with different configuration, then the last configuration wins. E.g.:

```
services
    .AddHttpClient<MyService1>()
    .CorrelateRequests(opts => opts.LoggingScopeKey = "ActivityId");

services
    .AddHttpClient<MyService2>()
    .CorrelateRequests(opts => opts.LoggingScopeKey = "CorrelationId");
```

With the above, it seems to me both clients will still use `CorrelationId` as log scope key. That could be resolved by using named options I suppose, but I also don't think configuring the log scope key belongs in the client registration at all anyway. Currently it is thus possible to set the log scope key at the top level and per client. This seems redundant, as you'd really only want to use one unique log scope key across the entire application.

As such, I believe it should only be configurable via an explicit call to `AddCorrelate()`. The fact that `AddCorrelate()` is called from within this registration method should not be relevant if one calls `AddCorrelate(opts => opts.LogScopeKey = "ActivityId")` before registering clients (due to the use of `services.TryAdd<>`).

This also removes the need to derive the client options from `CorrelationManagerOptions`.